thrust/thrust/system/tbb/detail/scan.inl

```diff
@@ -46,6 +46,100 @@ namespace detail
 namespace scan_detail
 {
 
+template <typename InputIterator, typename OutputIterator, typename BinaryFunction, typename ValueType>
+struct inclusive_body_init
+{
+  InputIterator input;
+  OutputIterator output;
+  thrust::detail::wrapped_function<BinaryFunction, ValueType> binary_op;
+  ValueType sum;
+  bool first_call;
+
+  inclusive_body_init(InputIterator input, OutputIterator output, BinaryFunction binary_op, ValueType init)
+      : input(input)
+      , output(output)
+      , binary_op{binary_op}
+      , sum(init)
+      , first_call(true)
+  {}
+
+  inclusive_body_init(inclusive_body_init& b, ::tbb::split)
+      : input(b.input)
+      , output(b.output)
+      , binary_op(b.binary_op)
+      , sum(b.sum)
```

Comment:
suggestion: if I remember correctly, split constructor runs concurrently while other thread is processing `b`. This might cause a race. Existing scan has it as well, so I'd suggest filing an issue to investigate this race later.

Reply:
#2277